src/tools/chdman.cpp

```diff
@@ -2591,66 +2551,155 @@ static void do_extract_cd(parameters_map &params)
 	if (output_file_str != params.end())
 		check_existing_output_file(params, *output_file_str->second);
 
-	// verify output BIN file doesn't exist
+	// determine output type based on the specified file extension
+	int mode = MODE_NORMAL;
+	if (core_filename_ends_with(*output_file_str->second, ".cue"))
+		mode = MODE_CUEBIN;
+	else if (core_filename_ends_with(*output_file_str->second, ".gdi"))
+		mode = MODE_GDI;
+
+	// determine the output bin filename based on provided input parameters
 	auto output_bin_file_fnd = params.find(OPTION_OUTPUT_BIN);
 	std::string default_name(*output_file_str->second);
+
+	// split path and extension
 	int chop = default_name.find_last_of('.');
 	if (chop != -1)
 		default_name.erase(chop, default_name.size());
-	std::string basename = default_name;
-	default_name.append(".bin");
+
+	// GDIs will always output as split bin
+	bool is_splitbin = mode == MODE_GDI || params.find(OPTION_OUTPUT_SPLIT) != params.end();
+	if (!is_splitbin && cdrom->is_gdrom() && mode == MODE_CUEBIN)
+	{
+		// GD-ROM cue/bin is in Redump format which should always be split by tracks
+		util::stream_format(std::cout, "Warning: --%s is required for this specific combination of input disc type and output format, enabling automatically\n", OPTION_OUTPUT_SPLIT);
+		is_splitbin = true;
+	}
+
+	if (is_splitbin)
+	{
+		if (mode == MODE_GDI)
+		{
+			default_name += "%02t";
+		}
+		else
+		{
+			const std::string format = toc.numtrks >= 10 ? "%02t" : "%t";
+			default_name += " (Track " + format + ")";
+		}
+	}
+
+	std::string output_bin_file_ext = ".bin";
 	std::string *output_bin_file_str;
 	if (output_bin_file_fnd == params.end())
+	{
 		output_bin_file_str = &default_name;
+	}
 	else
+	{
 		output_bin_file_str = output_bin_file_fnd->second;
 
-	check_existing_output_file(params, *output_bin_file_str);
+		chop = (*output_bin_file_str).find_last_of('.');
```

Comment:
You can use the `->` operator to avoid some ugliness – `chop = output_bin_file_str->find_last_of('.');` etc.  It’s present for pointers, iterators, and other similar things.

Reply:
Me completely forgetting how pointers work in C++. Fixed.